### PR TITLE
OF-793: Fixes for 'unsupported record version' exception

### PR DESCRIPTION
**Pull request by guusdk**
This isn't the nicest of fixes, but my tests started to fail after attempting to improve on this. As a result, I'm providing this as-is, as I suspect it's a functional improvement.

### CODE REVIEW

File: xmppserver/src/main/java/org/jivesoftware/openfire/net/TLSWrapper.java [hunk 1]
```diff
@@ -136,10 +139,21 @@ public void enableLogging(boolean logging) {
     public ByteBuffer unwrap(ByteBuffer net, ByteBuffer app) throws SSLException {
         ByteBuffer out = app;
         out = resizeApplicationBuffer(out);// guarantees enough room for unwrap
+
+        // Record a hex dump of the buffer, but only when logging on level 'debug'.
+        String hexDump = null;
+        if ( Log.isDebugEnabled() )
```
Comment (GregDThomas) [inline xmppserver/src/main/java/org/jivesoftware/openfire/net/TLSWrapper.java]: If debug is enabled, hexDump is calculated - but only ever used if the exception occurs. Is it worth moving this section to the exception handler?

File: xmppserver/src/main/java/org/jivesoftware/openfire/server/ServerDialback.java [hunk 2]
```diff
@@ -680,6 +680,9 @@ private VerifyResult sendVerifyKey(String key, StreamID streamID, String recipie
                 doc = reader.parseDocument();
             } catch (DocumentException e) {
                 log.warn("Unable to verify key: XML Error!", e);
+                // Close the stream
+                writer.write("</stream:stream>");
+                writer.flush();
```
Comment (GregDThomas) [inline xmppserver/src/main/java/org/jivesoftware/openfire/server/ServerDialback.java]:
```
writer.write("</stream:stream>");
writer.flush();
```
Appears a number of times in this file. I wonder if it's worth creating a utility method, `closeStreamAndFlush(writer)` so this is all captured in one place?

File: xmppserver/src/main/java/org/jivesoftware/openfire/net/TLSWrapper.java (hunk 1 above)
Comment (guusdk) [reply xmppserver/src/main/java/org/jivesoftware/openfire/net/TLSWrapper.java]: The intend is to capture the buffer before it gets touched (possibly modified) by the TLS handling.

Comment (GregDThomas) [reply xmppserver/src/main/java/org/jivesoftware/openfire/net/TLSWrapper.java]: A comment to that effect would probably help prevent this being refactored in error in the future ...

**Comment by GregDThomas** [comment]
Nothing major, just a few style comments, but seems OK.

File: xmppserver/src/main/java/org/jivesoftware/openfire/server/ServerDialback.java (hunk 2 above)
Comment (guusdk) [reply xmppserver/src/main/java/org/jivesoftware/openfire/server/ServerDialback.java]: That's not reducing complexity, that's just moving complexity around :)

**Pull request merged**